### PR TITLE
Deletion API code + APIController implementation

### PR DESCRIPTION
**Pull request by oshamoil**

### CODE REVIEW

File: backend/chromeApiServer/src/main/java/com/canvas/service/SubmissionDirectoryService.java [hunk 1]
```diff
@@ -105,6 +106,23 @@ private SubmissionFile[] writeSubmissionFiles(Map<String, byte[]> submissionFile
         return submissionFiles.toArray(new SubmissionFile[0]);
     }
 
+    public ResponseEntity<Deletion> deleteSubmissionDirectory(ExtensionUser user) {
+        String submissionDirectory = generateUniqueDirectoryName(user.getCourseId(), user.getAssignmentId(), user.getStudentId());
+        String description = user.getCourseId() + " " +  user.getAssignmentId() + " " + user.getStudentId() + " " + submissionDirectory;
+
+        // Now delete
+        fileService.deleteDirectory(submissionDirectory);
```
Comment (fkogge) [inline backend/chromeApiServer/src/main/java/com/canvas/service/SubmissionDirectoryService.java:114]: there's a helper method in this class that already calls fileService to delete the directory

File: backend/chromeApiServer/src/main/java/com/canvas/service/SubmissionDirectoryService.java [hunk 2]
```diff
@@ -105,6 +106,23 @@ private SubmissionFile[] writeSubmissionFiles(Map<String, byte[]> submissionFile
...
+
+        // TODO: FileService.deleteDirectory signature could be changed to boolean and plugged into success for builder
+        // if successful deletion
+        Deletion deletion = Deletion.builder()
+                .success(true)
```
Comment (fkogge) [inline backend/chromeApiServer/src/main/java/com/canvas/service/SubmissionDirectoryService.java:119]: can we add a case where the deletion is unsuccessful such as if the directory doesn't exist, and send back an error response?

**Review by fkogge — CHANGES_REQUESTED** [review]

**Comment by gaaliciA1990** [comment]
@oshamoil , please address the feedback by EOD today. Thanks

**Comment by oshamoil** [comment]
Will be resolved before EOD today @gaaliciA1990

File: backend/chromeApiServer/src/main/java/com/canvas/service/SubmissionDirectoryService.java (hunk 2 above)
Comment (oshamoil) [reply backend/chromeApiServer/src/main/java/com/canvas/service/SubmissionDirectoryService.java:119]: so in order to do this, from what I understand in the code and mentioned in the TODO above, we would have to change method signature of FileService.deleteDirectory from void too boolean. I don't think that will break anything but I will try and include it in some tests

**Pull request closed without merging**